### ISSUE TITLE
"this is undefined" on SingleDatePickerInput

### ISSUE THREAD

**Issue by 0xELJ** [airbnb/react-dates#1449]
Hi! I'm facing a problem with SingleDatePicker. My implementation for SingleDatePicker was working fine but now when I try to render the component I'm getting an error:

<img width="1010" alt="screen shot 2018-11-04 at 23 20 29" src="https://user-images.githubusercontent.com/37882104/47979336-44d84b80-e088-11e8-8ab6-f156f201ebe8.png">

The error is occurring in the SingleDatePickerInput.js file from the library. I was analyzing my code but I think it's fine, please take a look:

```jsx
import React, { Component } from 'react';
import { SingleDatePicker } from 'react-dates';
import validateBirthdate from '../utils/validateBirthdate';

class DatePicker extends Component {
  state = { focused: null };

  handleFocusChange({ focused }) {
    this.setState({ focused });
  }

  render() {
    const {
      label,
      placeholder,
      input: { value = null, onChange },
      meta: { error, touched }
    } = this.props;
    const { focused } = this.state;
    const invalidInput = touched && error;

    return (
      <div className={invalidInput ? 'invalid' : ''}>
        <label>{label}</label>
        <SingleDatePicker
          id="birthdatePicker"
          placeholder={placeholder}
          date={value}
          focused={focused}
          onDateChange={onChange}
          onFocusChange={this.handleFocusChange.bind(this)}
          isOutsideRange={validateBirthdate}
          showClearDate
          noBorder
          block
          hideKeyboardShortcutsPanel
        />
        {
          invalidInput &&
          <div className="invalid">
            <span>{error} <i className="material-icons left">do_not_disturb_on</i></span>
          </div>
        }
      </div>
    )
  }
}

export default DatePicker;
```

I'm working with the latest version (18.2.0) of the library.
What's wrong? Any suggestions?

**Comment by oskarom** [comment]
I'm running into this same bug, it appears to be introduced in version version 16.7.1 of the library.

**Comment by ljharb** [comment]
What version of react?

**Comment by omribar123** [comment]
I had the same bug, saw that there was a fix in v18.1.0 by Maja Wichrowska 
(commit e376d2fb9b27ee7dd8938881fd1a3dbc7d8fcdf8).

update the react-dates package and it will solve you the problem.
